src/DistributedWriteAheadLog/KafkaWALCommon.cpp

```diff
@@ -25,8 +26,24 @@ namespace DWAL
 {
 std::string escapeDWalName(const std::string & namespace_, const std::string & name_)
 {
-    /// FIXME : normalize name according to Kafka topic name "[a-zA-Z0-9\\._\\-]"
-    return namespace_ + "." + name_;
+    /// Allowed chars are ASCII alphanumerics, '.', '_' and '-'. '_' is used as escaped char in the form '_xx' where xx
```

Comment:
I would like to refactor this function as 

std::string escapeName(const std::string & s)
{
    std::string escaped;
    escaped.reserve(s.size());

    for (const auto & b : s)
    {
        if ((b >= 'a' && b <= 'z') || (b >= 'A' && b <= 'Z') || (b >= '0' && b <= '9') || (b == '.' || b == '-'))
        {
            s += b;
        }
        else
        {
            char out[3] = "_";
            writeHexByteUppercase(b, out + 1);
            s += out;
        }
    }
    return escaped;
}

std::string escapeDWalName(const std::string & namespace, const std::string & name)
{
    return escapeName(namespace) + "." + escapeName(name);
}

Reply:
fixed